File: Python/Validación_Número.py
def verificar_numero():
    print("Escriba un número para verificar si es válido: ")
    
    try:
        # Leer entrada del usuario
        entrada = input()
        numero = float(entrada)
        
        print("Es número")
        
        # Verificar tipo (entero o flotante)
        if numero.is_integer():
            tipo = "entero"
        else:
            tipo = "flotante"
        
        # Verificar signo
        if numero > 0:
            signo = "positivo"
        elif numero < 0:
            signo = "negativo"
        else:
            signo = "cero"
        
        # Verificar rango
        if 0 <= numero <= 100:
            rango = "dentro del rango 0-100"
        else:
            rango = "fuera del rango 0-100"
        
        # Mostrar resultados
        print(f"Es un {tipo} {signo}")
        
        if numero == 0:
            print("Su numero es 0")
        
        print(f"Está {rango}")
        
    except ValueError:
        print("No es un número")

# Ejecutar la función
if __name__ == "__main__":
    verificar_numero()

File: C/Validación_Número.c
#include <stdio.h>

float numero;

int main()
{
    printf("Escriba un número para verificar si es válido: \n");

    // Para verificar que sea un numero y recibimos el valor del scanf
    if (scanf("%f", &numero) == 1)
    {
        printf("Es número\n");
        // Para verificar que si es entero o flotante
        if (numero == (int)numero)
        {
            printf("Es un entero");
        }
        else
        {
            printf("Es un flotante");
        }
        // Para saber si es positivo o negativo
        if (numero > 0)
        {
            printf("Es positivo\n");
        }
        else if (numero < 0)
        {
            printf("Es negativo\n");
        }
        // Para verificar si es 0
        if (numero == 0)
        {
            printf("Su numero es 0\n");
        }
        // Para saber si está entre 0 y 100
        if (numero >= 0 && numero <= 100)
        {
            printf("Está entre 0 y 100\n");
        }
        else
        {
            printf("Fuera de rango entre 0 y 100\n");
        }
    }
    else
    {
        printf("No es un número\n");
    }
    return 0;
}